ostd/src/arch/x86/dispatcher.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{borrow::ToOwned, vec::Vec};
+
+use align_ext::AlignExt;
+
+use crate::{
+    boot::memory_region::MemoryRegionType, device::dispatcher::io_mem::IoMemDispatcherBuilder,
+};
+
+/// Initializes the allocatable MMIO area based on the x86-64 memory distribution map.
+///
+/// In x86-64, the available physical memory area is divided into two regions below 32 bits (Low memory)
+/// and above (High memory). Where the area from the top of low memory to 0xffff_ffff, after the top of
+/// high memory, is the available MMIO area.
```

Comment:
> Where the area from the top of low memory to 0xffff_ffff, after the top of high memory, is the available MMIO area.

I'm not sure if this is a legal English sentence, though. Please correct me if you think it is.

Otherwise, I guess we can just write:
>> The area from the top of low memory to 0xffff_ffff and the area after the top of high memory are available MMIO areas.